packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -332,13 +322,61 @@ export class TextFieldBase extends BaseComponent<ITextFieldProps, ITextFieldStat
     }
   }
 
+  private _getValueAndUpdateIsControlled(props: ITextFieldProps, isConstructor?: boolean): string {
```

Comment:
This function also seems fairly generic and I'm wondering if it can be pushed down into utils (possibly with the warned functions then encapsulated and not exported.) I was trying to think of a better name too, like `checkControlledValue`? Hmm..

Reply:
It's somewhat generic, but there are some challenges with moving it to utilities:
- Not all form components use the prop names `value` and `defaultValue`, so we'd have to add either the prop names or prop values as params
- Would need to add param for previous `isControlled` value and current state value
- Would need to return an object with the new value and controlled state

Obviously these can all be resolved, question is just whether it's worth doing so (and how).